interfaces/cython/cantera/test/test_reaction.py

```diff
@@ -1490,6 +1491,22 @@ def test_custom_lambda(self):
         rxn = self.from_rate(lambda T: 38.7 * T**2.7 * exp(-3150.15428/T))
         self.check_rxn(rxn)
 
+    def test_persistent(self):
+        # ensure that temporary Python objects are buffered
+        gas = ct.Solution(thermo=self.soln.thermo_model,
+                         kinetics=self.soln.kinetics_model,
+                         species=self.species, reactions=[])
+        for i in range(5):
+            R = ct.Reaction(equation='H + OH <=> H2O', rate=ct.CustomRate(lambda T: T))
+            gas.add_reaction(R)
+            R = ct.Reaction(equation='H + OH <=> H2O', rate=lambda T: T)
+            gas.add_reaction(R)
+            def func(T):
+                return T
+            R = ct.Reaction(equation='H + OH <=> H2O', rate=func)
+            gas.add_reaction(R)
+        assert (gas.forward_rate_constants == gas.T).all()
```

Comment:
@bryanwweber ... this is the crux of the issue ... `R`'s are temporary - see also @speth's example in the #1282